src/logic/handler.js

```diff
@@ -0,0 +1,37 @@
+const assert = require('assert');
+const { asApiGatewayResponse } = require('../response');
+
+module.exports.wrap = (handler, {
+  request,
+  route,
+  router,
+  module,
+  params = [],
+  options = {}
+}) => async (event, context) => {
+  if (!event.httpMethod) {
+    return Promise.resolve('OK - No API Gateway call detected.');
+  }
+  const kwargs = {
+    request,
+    event,
+    context,
+    route,
+    router,
+    params,
+    options
+  };
+  const response = await module.before(kwargs)
+    .then(() => handler(event.parsedParameters, context, event))
```

Comment:
`then` chaining an `await` is kind of an anti-pattern, did you want to separate this a bit?